@here/harp-mapview/lib/text/TextElementState.ts

```diff
@@ -86,8 +144,57 @@ export class TextElementState {
         return this.m_viewDistance;
     }
 
-    set viewDistance(distance: number | undefined) {
-        this.m_viewDistance = distance;
+    /**
+     * Sets the distance of the element to the current view center.
+     * @param viewDistance The new view distance to set. If `undefined`, element is considered to
+     * be out of view.
+     * @param groupState The state of the group the element belongs to.
+     */
+    setViewDistance(viewDistance: number | undefined, groupState: TextElementGroupState) {
+        if (viewDistance === this.m_viewDistance) {
+            return;
+        }
+        this.m_viewDistance = viewDistance;
+        if (this.m_viewDistance !== undefined) {
+            groupState.invalidateSorting();
```

Comment:
setViewDistance is not only called from TextElementGroupState, but also from the TextElementsRenderer during label placement, since the distance must be updated on every frame to have smooth scaling.